dbos/_dbos.py

```diff
@@ -186,6 +188,20 @@ def register_instance(self, inst: object) -> None:
         else:
             self.instance_info_map[fn] = inst
 
+    def compute_app_version(self) -> str:
+        """
+        An application's version is computed from a hash of the source of its workflows.
+        This is guaranteed to be stable given identical source code because it uses an MD5 hash
+        and because it iterates through the workflows in insertion order (which Python dicts guarantee).
+        This way, if the app's workflows are updated (which would break recovery), its version changes.
+        App version can be manually set through the DBOS__APPVERSION environment variable.
+        """
+        hasher = hashlib.md5()
+        for wf in self.workflow_info_map.values():
+            source = inspect.getsource(wf)
+            hasher.update(source.encode("utf-8"))
+        return hasher.hexdigest()
+
```

Comment:
It would be better to manually sort the source code strings, rather than relying on insertion order.

If I change my app by just placing workflow 2 before workflow 1 -- maybe for some sort of docs purpose, whatever,

I don't want the version to change (because the source code didn't change)

I'm told this would work

```
hasher = hashlib.md5()
sources = sorted(inspect.getsource(wf) for wf in workflow_info_map.values())
for source in sources:
    hasher.update(source.encode("utf-8"))
return hasher.hexdigest()
```

Reply:
That said, we should probably sort them anyways over stability in import order concerns.